src/scripts/slurm_submit.sh

```diff
@@ -85,6 +85,14 @@ if [[ $bls_opt_mpinodes -gt 1 ]] ; then
   echo "#SBATCH --cpus-per-task=$bls_opt_mpinodes" >> $bls_tmp_file
 fi
 
+# add GPU support
+[ -z "$bls_opt_gpunumber" ] || [ -z "$bls_opt_gpumodel" ] || echo "#SBATCH --gres=gpu:${bls_opt_gpumodel}:${bls_opt_gpunumber}" >> $bls_tmp_file
+[ -z "$bls_opt_gpunumber" ] || [ ! -z "$bls_opt_gpumodel" ] || echo "#SBATCH --gres=gpu:${bls_opt_gpunumber}" >> $bls_tmp_file
+[ ! -z "$bls_opt_gpunumber" ] || [ -z "$bls_opt_gpumodel" ] || echo "#SBATCH --gres=gpu:${bls_opt_gpumodel}:1" >> $bls_tmp_file
```

Comment:
Just a guess, but i suspect the logic might be wrong here, with all the extra negations.

I'd bet a beer, actually.

I'd suggest using `if` blocks or `&&` for these conditions -- it's kind of easy to get wrong otherwise.  And it's not easy to follow, reading it, either.

Reply:
Ah, after closer inspection, maybe it is correct.  But it is a bit painful to try to make sense of it.

Reply:
Agreed on the confusing syntax. Barring bugs, I'd prefer to leave it alone to avoid divergence from upstream.